tests/unittests/libretroshare/gxs/common/data_support.cc

```diff
@@ -209,10 +213,11 @@ RsSerialType* init_item(RsNxsSyncGrpItem& rsgl)
     rsgl.publishTs = rand()%23;
     init_random(rsgl.grpId) ;
 
-    return new RsNxsSerialiser(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
+    if(ser)
+    *ser = new RsNxsSerialiser(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
 }
 
-RsSerialType* init_item(RsNxsSyncMsgItem& rsgml)
+void init_item(RsNxsSyncMsgItem& rsgml,RsSerialType **ser)

```

Comment:
two reasons: there was a leak in the unittest due to some code parts not using the return value of these, and also for consistency reasons: the function  that inits an item should not create memory at the same time. I understand that this had been done because there was no way in the calling template functions to know  the type of the serialiser. But I also changed it so it's not needed anymore.